### PR TITLE
Ci set concretiztion pool size

### PR DESCRIPTION
**Pull request by kwryankrattiger**
Take two on specifying concretization for generation. Moved the request from 4 -> 8 in this version.

CC: @tldahlgren @alalazo

### CODE REVIEW

File: lib/spack/spack/cmd/ci.py [hunk 1]
```diff
@@ -119,6 +120,9 @@ def setup_parser(subparser):
     )
     generate.set_defaults(func=ci_generate)
 
+    spack.cmd.common.arguments.add_concretizer_args(generate)
+    spack.cmd.common.arguments.add_common_arguments(generate, ["jobs"])
```
Comment (scottwittenburg) [inline lib/spack/spack/cmd/ci.py:124]: For my education, can you explain how this `-j` value ends up affecting the number of processes used for concretization?  I see how you've used it for `rebuild`, just wondering why something similar isn't necessary for `generate`.

Comment (kwryankrattiger) [reply lib/spack/spack/cmd/ci.py:124]: the config `config:build_jobs` gets set in `lib/spack/spack/cmd/common/arguments.py:SetParallelJobs` by the common `-j`/`--jobs` argument.

Then, concretization uses the `build_jobs` config parameter to decide how many concretization tasks to spawn for concretize seperately.

The call stack is `lib/spack/spack/concretize.py` calls `spack.config.determine_number_of_jobs` and then queries the config for `config:build_jobs` or a default of 16.

Comment (scottwittenburg) [reply lib/spack/spack/cmd/ci.py:124]: ~How does `config:build_jobs` get set?~ Oh, I've just seen the `arg` decorator in there that populates `_arguments`, which is referenced by `add_common_arguments`.  Thanks.

**Comment by kwryankrattiger** [comment]
Looking at the concretization times for this PR, of all of the stacks we are maxing out around 20 minutes to concretize (e4s). That should be enough head room for the 1hr timeout.

**Comment by kwryankrattiger** [comment]
@mvandenburgh @zackgalbreath @johnwparent any ideas on why windows runners are unable to be picked up? Looking at the configs we have the requests are still way under the normal threshold but maybe I am missing something.

**Comment by alalazo** [comment]
I think the slowest is the `e4s-power` (as expected): https://gitlab.spack.io/spack/spack/-/jobs/14191928 33 minutes

**Comment by kwryankrattiger** [comment]
> I think the slowest is the e4s-power

This one already had exceptions for concretization, but yeah, it is the slowest pretty consistently (~25-35 minutes)

**Comment by kwryankrattiger** [comment]
Verified rebuild and generate (https://gitlab.spack.io/spack/spack/-/jobs/14188046)

**Comment by alalazo** [comment]
@spackbot run pipeline (Win runner system failure)

**Review by zackgalbreath — APPROVED** [review]

**Pull request merged**